File: 2. Stack/Infix Prefix Postfix/06. Prefix to Postfix/PrefixPostfix.java
import java.util.*;

public class PrefixPostfix {

	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		String prefix_exp="";
		System.out.println("Enter prefix Expression");
		prefix_exp=sc.next();
		System.out.println("Equivalent postfix Expression for "+prefix_exp+" is= "+PrefixToPostfix(prefix_exp));

	}

	private static String PrefixToPostfix(String prefix_exp) {
		Stack<String> st=new Stack();
		
		for(int i=prefix_exp.length()-1;i>=0;i--) {
			char ch=prefix_exp.charAt(i);
			if((ch >= 'a' && ch <= 'z') ||(ch >= 'A' && ch <= 'Z')) {
				st.push(Character.toString(ch));
			}
			else {
				String op1=st.pop();
				String op2=st.pop();
				String exp=op1+op2+ch;
				st.push(exp);
			}
		}

		return st.peek();
	}

}

File: 01. DataStructures/02. Stack/Infix Prefix Postfix/06. Prefix to Postfix/PrefixPostfix.cpp
#include <iostream>
#include <string>
#include <stack>
#include <unordered_set>

std::string prefixToPostfix(std::string prefix)
{
    std::stack<std::string> prefixPostfixStack;
    std::unordered_set<char> symbols = {'+', '-', '*', '/', '^'};

    int n = prefix.size();

    for (int i = n - 1; i >= 0; i--)
    {
        if (symbols.find(prefix[i]) != symbols.end())
        {
            std::string op1 = prefixPostfixStack.top();
            prefixPostfixStack.pop();
            std::string op2 = prefixPostfixStack.top();
            prefixPostfixStack.pop();

            std::string t = op1 + op2 + prefix[i];
            
            prefixPostfixStack.push(t);
        }
        else
        {
            std::string t = std::string(1, prefix[i]);
            prefixPostfixStack.push(t);
        }
    }
    return prefixPostfixStack.top();
}

int main(){

    std::string prefix;
    std::cout << "Enter the prefix expression: ";
    std::cin >> prefix;

    std::cout << "The Postfix expression of the entered Prefix expression is: " << prefixToPostfix(prefix);

    return 0;
}